### PR TITLE
ESQL: Make EvalBenchmarks executable again

### PR DESCRIPTION
**Pull request by alex-spies**

### CODE REVIEW

File: benchmarks/src/main/java/org/elasticsearch/benchmark/compute/operator/EvalBenchmark.java [hunk 1]
```diff
@@ -61,6 +61,11 @@
 public class EvalBenchmark {
     private static final int BLOCK_LENGTH = 8 * 1024;
 
+    static final DriverContext driverContext = new DriverContext(
+        BigArrays.NON_RECYCLING_INSTANCE,
+        BlockFactory.getInstance(new NoopCircuitBreaker("noop"), BigArrays.NON_RECYCLING_INSTANCE)
+    );
+
     static {
```
Comment (alex-spies) [inline benchmarks/src/main/java/org/elasticsearch/benchmark/compute/operator/EvalBenchmark.java:69]: This needs to be loaded first before it's used by calling `run` in the `static` block below.

File: benchmarks/src/main/java/org/elasticsearch/benchmark/compute/operator/EvalBenchmark.java [hunk 2]
```diff
@@ -27,11 +27,11 @@
 import org.elasticsearch.xpack.esql.expression.function.scalar.date.DateTrunc;
 import org.elasticsearch.xpack.esql.expression.function.scalar.math.Abs;
 import org.elasticsearch.xpack.esql.expression.function.scalar.multivalue.MvMin;
+import org.elasticsearch.xpack.esql.expression.predicate.operator.arithmetic.Add;
 import org.elasticsearch.xpack.esql.planner.Layout;
 import org.elasticsearch.xpack.esql.type.EsqlDataTypes;
 import org.elasticsearch.xpack.ql.expression.FieldAttribute;
 import org.elasticsearch.xpack.ql.expression.Literal;
-import org.elasticsearch.xpack.ql.expression.predicate.operator.arithmetic.Add;
 import org.elasticsearch.xpack.ql.tree.Source;
```
Comment (alex-spies) [inline benchmarks/src/main/java/org/elasticsearch/benchmark/compute/operator/EvalBenchmark.java:35]: Wrong `Add` :D

File: x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java [hunk 3]
```diff
@@ -273,58 +271,6 @@ private PhysicalOperation planFieldExtractNode(LocalExecutionPlannerContext cont
         return physicalOperationProviders.fieldExtractPhysicalOperation(fieldExtractExec, plan(fieldExtractExec.child(), context));
     }
 
-    /**
-     * Map QL's {@link DataType} to the compute engine's {@link ElementType}.
-     */
-    public static ElementType toElementType(DataType dataType) {
```
Comment (alex-spies) [inline x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/LocalExecutionPlanner.java:279]: This was a fun one. `LocalExecutionPlanner`'s class calls `LogManager::getLogger` which gives a NPE during benchmarks. Moving to `PlannerUtils` solves this.

File: benchmarks/src/main/java/org/elasticsearch/benchmark/compute/operator/EvalBenchmark.java (hunk 2 above)
Comment (nik9000) [reply benchmarks/src/main/java/org/elasticsearch/benchmark/compute/operator/EvalBenchmark.java:35]: oof.

**Review by nik9000 — APPROVED** [review]
Wow.

I really wish we ran the benchmarks on every PR. Not, like, completely. But maybe just run them in some kind of test mode. Just so we know they still run! But that's a problem for another time.

**Pull request merged**